KSCrash.podspec

```diff
@@ -16,123 +16,68 @@ Pod::Spec.new do |s|
-        sets.dependency 'KSCrash/Reporting/Filters/Stringify'
-        sets.dependency 'KSCrash/Reporting/Filters/GZip'
-        sets.dependency 'KSCrash/Reporting/Filters/JSON'
-
-        sets.source_files = 'Source/KSCrash/Reporting/Filters/KSCrashReportFilterSets.h',
-                            'Source/KSCrash/Reporting/Filters/KSCrashReportFilterSets.m'
-      end
-
-      filters.subspec 'Tools' do |tools|
-        tools.source_files = 'Source/KSCrash/Reporting/Filters/Tools/**/*.{h,m,mm,c,cpp}'
-      end
-
-    end
-
-    reporting.subspec 'Tools' do |tools|
-      tools.ios.frameworks = 'SystemConfiguration'
-      tools.tvos.frameworks = 'SystemConfiguration'
-      tools.osx.frameworks = 'SystemConfiguration'
-      tools.source_files = 'Source/KSCrash/Reporting/Tools/**/*.{h,m,mm,c,cpp}',
-                           'Source/KSCrash/Recording/KSSystemCapabilities.h'
-    end
-
-    reporting.subspec 'MessageUI' do |messageui|
-    end
-
-    reporting.subspec 'Sinks' do |sinks|
-      sinks.ios.frameworks = 'MessageUI'
-      sinks.dependency 'KSCrash/Reporting/Filters'
-      sinks.dependency 'KSCrash/Reporting/Tools'
-      sinks.source_files = 'Source/KSCrash/Reporting/Sinks/**/*.{h,m,mm,c,cpp}'
-    end
+  s.subspec 'Filters' do |filters|
+    filters.dependency 'KSCrash/Recording'
+    filters.dependency 'KSCrash/RecordingCore'
+    filters.dependency 'KSCrash/ReportingCore'
 
+    configure_subspec.call(filters)
+  end
+
+  s.subspec 'Sinks' do |sinks|
+    sinks.dependency 'KSCrash/Recording'
+    sinks.dependency 'KSCrash/Filters'
+
+    configure_subspec.call(sinks)
   end
 
   s.subspec 'Installations' do |installations|
+    installations.dependency 'KSCrash/Filters'
+    installations.dependency 'KSCrash/Sinks'
     installations.dependency 'KSCrash/Recording'
-    installations.dependency 'KSCrash/Reporting'
-    installations.source_files = 'Source/KSCrash/Installations/**/*.{h,m,mm,c,cpp}'
+
+    configure_subspec.call(installations)
+  end
+
+  s.subspec 'RecordingCore' do |recording_core|
+    recording_core.dependency 'KSCrash/Core'
+    recording_core.dependency 'KSCrash/Swift'
+    recording_core.dependency 'KSCrash/LLVM'
+
+    configure_subspec.call(recording_core)
+  end
+
+  s.subspec 'ReportingCore' do |reporting_core|
+    reporting_core.dependency 'KSCrash/Core'
+    reporting_core.dependency 'KSCrash/Swift'
+    reporting_core.dependency 'KSCrash/LLVM'
+
+    configure_subspec.call(reporting_core)
   end
 
   s.subspec 'Core' do |core|
-    core.dependency 'KSCrash/Reporting/Filters/Basic'
-    core.source_files = 'Source/KSCrash/Installations/KSCrashInstallation.h',
-                        'Source/KSCrash/Installations/KSCrashInstallation.m',
-                        'Source/KSCrash/Installations/KSCrashInstallation+Private.h',
-                        'Source/KSCrash/Reporting/Tools/KSCString.{h,m}'
+    configure_subspec.call(core)
   end
 
+  s.subspec 'Swift' do |swift|
```

Comment:
Have an issue with `.def` files during `pod lib lint`. Fails with `--use-modular-headers`, `--use-static-frameworks` and standard dynamic frameworks. Compiles fine with `--use-libraries`.

```log
CompileC <$HOME>/Developer/Xcode/DerivedData/KSCrashPodIntegration-erekorvvqreivicfxtwmojmoxejn/Build/Intermediates.noindex/Pods.build/Debug-iphonesimulator/KSCrash.build/Objects-normal/arm64/ManglingUtils.o <$HOME>/Developer/KSCrashPodIntegration/Pods/KSCrash/Sources/KSCrashSwift/ManglingUtils.cpp normal arm64 c++ com.apple.compilers.llvm.clang.1_0.compiler (in target 'KSCrash' from project 'Pods')
    cd <$HOME>/Developer/KSCrashPodIntegration/Pods
    /Applications/Xcode-15.3.0.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -x c++ -ivfsstatcache <$HOME>/Developer/Xcode/DerivedData/SDKStatCaches.noindex/iphonesimulator17.4-21E210-e0f8f310ab6fbd8ae890b8f8b8ba93cb.sdkstatcache -fmessage-length\=0 -fdiagnostics-show-note-include-stack -fmacro-backtrace-limit\=0 -fno-color-diagnostics -fmodules-prune-interval\=86400 -fmodules-prune-after\=345600 -fbuild-session-file\=<$HOME>/Developer/Xcode/DerivedData/ModuleCache.noindex/Session.modulevalidation -fmodules-validate-once-per-build-session -Wnon-modular-include-in-framework-module -Werror\=non-modular-include-in-framework-module -Wno-trigraphs -Wno-missing-field-initializers -Wno-missing-prototypes -Werror\=return-type -Wdocumentation -Wunreachable-code -Werror\=deprecated-objc-isa-usage -Werror\=objc-root-class -Wno-non-virtual-dtor -Wno-overloaded-virtual -Wno-exit-time-destructors -Wno-missing-braces -Wparentheses -Wswitch -Wunused-function -Wno-unused-label -Wno-unused-parameter -Wunused-variable -Wunused-value -Wempty-body -Wuninitialized -Wconditional-uninitialized -Wno-unknown-pragmas -Wno-shadow -Wno-four-char-constants -Wno-conversion -Wconstant-conversion -Wint-conversion -Wbool-conversion -Wenum-conversion -Wno-float-conversion -Wnon-literal-null-conversion -Wobjc-literal-conversion -Wshorten-64-to-32 -Wno-newline-eof -Wno-c++11-extensions -Wno-implicit-fallthrough -fstrict-aliasing -Wdeprecated-declarations -Winvalid-offsetof -Wno-sign-conversion -Winfinite-recursion -Wmove -Wcomma -Wblock-capture-autoreleasing -Wstrict-prototypes -Wrange-loop-analysis -Wno-semicolon-before-method-body -Wunguarded-availability -index-store-path <$HOME>/Developer/Xcode/DerivedData/KSCrashPodIntegration-erekorvvqreivicfxtwmojmoxejn/Index.noindex/DataStore @<$HOME>/Developer/Xcode/DerivedData/KSCrashPodIntegration-erekorvvqreivicfxtwmojmoxejn/Build/Intermediates.noindex/Pods.build/Debug-iphonesimulator/KSCrash.build/Objects-normal/arm64/82b82416624d2658e5098eb0a28c15c5-common-args.resp -include <$HOME>/Developer/KSCrashPodIntegration/Pods/Target\ Support\ Files/KSCrash/KSCrash-prefix.pch -MMD -MT dependencies -MF <$HOME>/Developer/Xcode/DerivedData/KSCrashPodIntegration-erekorvvqreivicfxtwmojmoxejn/Build/Intermediates.noindex/Pods.build/Debug-iphonesimulator/KSCrash.build/Objects-normal/arm64/ManglingUtils.d --serialize-diagnostics <$HOME>/Developer/Xcode/DerivedData/KSCrashPodIntegration-erekorvvqreivicfxtwmojmoxejn/Build/Intermediates.noindex/Pods.build/Debug-iphonesimulator/KSCrash.build/Objects-normal/arm64/ManglingUtils.dia -c <$HOME>/Developer/KSCrashPodIntegration/Pods/KSCrash/Sources/KSCrashSwift/ManglingUtils.cpp -o <$HOME>/Developer/Xcode/DerivedData/KSCrashPodIntegration-erekorvvqreivicfxtwmojmoxejn/Build/Intermediates.noindex/Pods.build/Debug-iphonesimulator/KSCrash.build/Objects-normal/arm64/ManglingUtils.o -index-unit-output-path /Pods.build/Debug-iphonesimulator/KSCrash.build/Objects-normal/arm64/ManglingUtils.o

<$HOME>/Developer/KSCrashPodIntegration/Pods/KSCrash/Sources/KSCrashSwift/ManglingUtils.cpp:75:10: fatal error: 'StandardTypesMangling.def' file not found
#include "StandardTypesMangling.def"
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
```

Reply:
Thanks for making these forks private